sdk/samples/router/general.py

```diff
@@ -0,0 +1,265 @@
+                )
+            ],
+            routes=[
+                'meow',
+                'woof',
+                'baaa',
+                'oink',
+                'ring-ding-ding'
+            ]
+        )
+    ]
+
+    # # Create an experiment config (
+    experiment_config = ExperimentConfig(
+        type="xp",
+        config={
+            'variables':
+                [
+                    {'name': 'farm_id', 'field': 'farm_id', 'field_source': 'header'},
+                    {'name': 'country_code', 'field': 'country', 'field_source': 'header'},
+                    {'name': 'latitude', 'field': 'farm_lat', 'field_source': 'header'},
+                    {'name': 'longitude', 'field': 'farm_long', 'field_source': 'header'}
+                ],
+            'project_id': 102
+        }
+    )
+
+    # # Create a resource request config for the router
+    resource_request = ResourceRequest(
+        min_replica=0,
+        max_replica=2,
+        cpu_request="500m",
+        memory_request="512Mi"
+    )
+
+    # # Create a log config for the router
+    log_config = LogConfig(
+        result_logger_type=ResultLoggerType.NOP
+    )
+
+    # # Create an enricher for the router
+    enricher = Enricher(
+        image="asia.gcr.io/gods-dev/echo:1.0.2",
+        resource_request=ResourceRequest(
+            min_replica=0,
+            max_replica=2,
+            cpu_request="500m",
+            memory_request="512Mi"
+        ),
+        endpoint="/",
+        timeout="60ms",
+        port=8080,
+        env=[
+            EnvVar(
+                name="humans",
+                value="farmer-joe"
+            )
+        ]
+    )
+
+    # # Create an ensembler for the router
+    ensembler = DockerRouterEnsemblerConfig(
+        id=1,
+        image="asia.gcr.io/gods-dev/echo:1.0.2",
+        resource_request=ResourceRequest(
+            min_replica=1,
+            max_replica=3,
+            cpu_request="500m",
+            memory_request="512Mi"
+        ),
+        endpoint="/echo",
+        timeout="60ms",
+        port=8080,
+        env=[],
+    )
+
+    # # Create the RouterConfig instance
+    router_config = RouterConfig(
+        environment_name="id-dev",
+        name="what-does-the-fox-say-1",
+        routes=routes,
+        rules=rules,
+        default_route_id="test",
+        experiment_engine=experiment_config,
+        resource_request=resource_request,
+        timeout="100ms",
+        log_config=log_config,
+        enricher=enricher,
+        ensembler=ensembler
+    )
+
+    # 1. Create a new router using the RouterConfig object
+    new_router = turing.Router.create(router_config)
+
+    # 2. List all routers
+    routers = turing.Router.list()
+    for r in routers:
+        if r.name == new_router.name:
+            my_router_id = r.id
+        print(r)
+
+    # 3. Get the router you just created using the router_id obtained
+    my_router = turing.Router.get(my_router_id)
```

Comment:
Some of these steps require waiting for the steps above to 'complete' before they can be run. For example, the one above (step 3) requires the router created in step 1 to be fully deployed before it's able to retrieve a non-`None` router config. I was wondering if I should write the sample in a notebook so that it's clearer that these steps should be run incrementally instead of a continuous script.

Reply:
Alternatively, you could consider writing a generic function which uses `time.sleep` to check whether router has been fully deployed prior to continuing the script.

Reply:
Ya I agree with @terryyylim you could write a blocking function to ensure that the router has been deployed and expose it as part of an Router method.

Reply:
Ok oops I think I might not have explained it carefully, but this file is really just meant as a sample, so the methods like `get` above wouldn't normally require waiting for the router to be deployed before being able to run (I might imagine someone wanting to get a router that's still `pending`).

What's particular in this sample is that I've laid out the various SDK functions in a certain order so that it demonstrates to the user the creation of *a* router, followed by the retrieval of *that* router by its id, and then the updating of *that* router, etc. So it just so happens that in this 'demo', this `get` should ideally retrieve a router that's already deployed because the next SDK function in the demo is one that requires it (or some other router whose existence I can't guarantee when the user runs the sample) to be deployed, which isn't really the case in normal usage.

Reply:
But yes alternatively, it's possible to implement some blocking function into these SDK methods, but that'll fundamentally change the way they operate, e.g. `get` would only work (or would at least wait until) on routers that are `deployed`, instead of being able to retrieve routers on any kind of status (`pending`, `undeployed`, or `failed`)

Reply:
> But yes alternatively, it's possible to implement some blocking function into these SDK methods, but that'll fundamentally change the way they operate,

You don't have to bake it into the existing methods, you can create a new method and that method only blocks and does nothing else.

Reply:
Ahh that looks elegant enough for the sample, alright I'll implement that :) Thanks guys!